Aplib.Tests/Core/Belief/CircularArrayTests.cs

```diff
@@ -0,0 +1,112 @@
+using Aplib.Core;
+
+namespace Aplib.Tests.Core.Belief;
+
+public class CircularArrayTests
+{
+    /// <summary>
+    /// Given a CircularArray instance,
+    /// When an element is put into the array,
+    /// The array should wrap around when it reaches its end.
+    /// (i.e., the first element should become the previous second element, 
+    /// and the last element should be the new element)
+    /// </summary>
+    [Fact]
+    public void Put_ArrayIsFull_UpdatesHead() 
+    {
+        // Arrange
+        CircularArray<int> circularArray = new([1, 2, 3]);
+
+        // Act
+        circularArray.Put(4);
+
+        // Assert
+        Assert.Equal(2, circularArray[0]);
+        Assert.Equal(3, circularArray[1]);
+        Assert.Equal(4, circularArray[2]);
+    }
+
+    /// <summary>
+    /// Given a CircularArray instance,
+    /// When the head is updated,
+    /// Setting an element should set the correct index 
+    /// even if the head is not at the start of the array.
+    /// </summary>
+    [Fact]
+    public void Set_HeadIsUpdated_SetsCorrectIndex()
+    {
+        // Arrange
+        CircularArray<int> circularArray = new(3);
+
+        // Act
+        circularArray[1] = 4;
+        circularArray.Put(6);
+        circularArray[1] = 5;
```

Comment:
Maybe add comments on what exactly is happening here? For example
```suggestion
        // circularArray.ToArray() == [0, 0, 0]
        circularArray[1] = 4;
        // circularArray.ToArray() == [0, 4, 0]
        circularArray.Put(6);
        // circularArray.ToArray() == [4, 0, 6]
        circularArray[1] = 5;
        // circularArray.ToArray() == [4, 5, 6]
```